libdrgn/kdump_reader.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 - Serapheim Dimitropoulos
+// SPDX-License-Identifier: GPL-3.0+
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "kdump_reader.h"
+#include "program.h"
+
+bool has_kdump_signature(int fd, struct drgn_error **err)
+{
+	char signature[SIG_LEN];
+
+	int n = read(fd, signature, sizeof (signature));
+	if (n != sizeof (signature)) {
+		*err = drgn_error_format(DRGN_ERROR_FAULT,
+		                         "short read from file");
```

Comment:
This needs the usual handling for short reads and `EINTR`. And, this should use `drgn_error_create_os()`.